code/game/objects/items/devices/scanners.dm

```diff
@@ -275,7 +275,7 @@ MASS SPECTROMETER
 			to_chat(user, "<span class='notice'>[cyberimp_detect]</span>")
 
 /proc/chemscan(mob/living/user, mob/living/M)
-	if(ishuman(M))
+	if(ismob(M))
 		var/mob/living/carbon/human/H = M

```

Comment:
you're playing a dangerous game

Reply:
i believe iscarbon is a check, should use that and cut the typecasting back to carbon

Reply:
I initially used iscarbon, but simple_animals are not carbon and it kinda felt better if they worked too. And they do, I've actually tested them as I said. The reagents don't do much for them but it's not like the health analyzer cares about that.

Reply:
use ```isliving()``` not ```ismob()```, living will cover simples and carbons, where as mob covers ghosts, AI camera eyes, and all sorts of DONT LOOK AT THESE mobs.

then cut the casting back to just ```var/mob/living/L = M```

Reply:
there's no need to cast when M is supposed to be living already

Reply:
yes, that's what i meant, and is fine.